activity/s3/activity.go

```diff
@@ -64,8 +68,16 @@ func (a *Activity) Eval(ctx activity.Context) (done bool, err error) {
 		return false, err
 	}
 
+	logger = ctx.Logger()
+	var action string
+	if a.settings.Action != "" {
```

Comment:
The logic here is incorrect.  The runtime input always overrides the setting if it is not empty.  It should be something like:
``` 
action := a.settings.Action
if in.Action != "" {
    action = in.Action 
}
```